enhancements/cluster-api/installing-openshift-natively-via-cluster-api.md

```diff
@@ -0,0 +1,598 @@
+For example, on AWS, where Cluster API does not manage DNS, a supplemental provider will be created to manage DNS records for the cluster.
+
+Provide a configurable, opinionated and controller driven multi-cluster and single cluster provisioning experience:
+* Driven by install config and Cluster API resources
+* Cluster API resources generated by the installer for opinionated install
+* Self crafted Cluster API resources for custom install or install on external platforms
+* Customisation of bootstrap manifests via control plane provider configuration
+
+### Workflow Description
+
+#### Opinionated installer generated infrastructure definitions
+
+The installer binary will be updated to transform the existing install config into Cluster API resources.
+
+During the `manifests` phase, the installer will generate the following Cluster API resources into an `infrastructure-manifests` folder:
+* Cluster
+* OpenShiftControlPlane
+* InfrastructureCluster (e.g. `AWSCluster`)
+* InfrastructureMachineTemplate (e.g. `AWSMachineTemplate`)
+
+Additionally supplemental resources will be required:
+* Namespace (each cluster will be created in a unique namespace for isolation)
+* Infrastructure Credentials secret (to allow the Cluster API control plane to authenticate to the cloud provider).
+
+The installer will then diverge from the current path, which is to execute `ignition-configs` and then `cluster` phases.
+
+The `ignition-configs` phase will become a terminal phase and will be used by the bootstrap provider to generate the ignition configs for the bootstrap node, control plane nodes and worker nodes.
+No changes are expected to the `ignition-configs` phase and existing UPI workflows should continue to work.
+
+The `cluster` phase will now skip the `ignition-configs` phase and will instead apply the Cluster API resources generated in the `manifests` phase to the Cluster API control plane.
+
+The installer will directly apply the Cluster API resources to the Cluster API control plane.
+If any modifications are made between the `manifests` and `cluster` phases, the installer will not overwrite these changes.
+This allows the user to tweak the configuration generated by the installer should they wish to do so.
+
+The installer will also create an `install-state` secret in the cluster namespace containing the current `.openshift_install_state.json` file.
+This will then be consumed later by the bootstrap provider to generation ignition.
+This secret will be referenced in the `OpenShiftControlPlane` spec.
+
+To allow the user to customise manifests, the installer will take all manifests from the `manifests` and `openshift` folders and wrap them into secrets to be applied to the cluster namespace.
+Each secret will be annotated to indicate that it should be included in the ignition generation phase, and to identify whether it was a `manifest` file or `openshift` file.
```

Comment:
Correct, but I don't think we do any validation there today, so I think that is a pre-existing problem. The installer currently loads the files from disk and puts them into the bootstrap ignition directly, so if any file has been edited and is malformed today, it would result in the same UX